packages/gatsby-source-shopify/src/gatsby-node.ts

```diff
@@ -0,0 +1,284 @@
+import { createOperations } from "./operations"
+import { eventsApi } from "./events"
+import {
+  CreateResolversArgs,
+  NodePluginArgs,
+  PluginOptionsSchemaArgs,
+  SourceNodesArgs,
+} from "gatsby"
+import { makeResolveGatsbyImageData } from "./resolve-gatsby-image-data"
+import {
+  getGatsbyImageResolver,
+  IGatsbyGraphQLResolverArgumentConfig,
+} from "gatsby-plugin-image/graphql-utils"
+import { shiftLeft } from "shift-left"
+import { pluginErrorCodes as errorCodes } from "./errors"
+import { makeSourceFromOperation } from "./make-source-from-operation"
+export { createSchemaCustomization } from "./create-schema-customization"
+import { createNodeId } from "./node-builder"
+import { JoiObject } from "joi"
+
+export function pluginOptionsSchema({
+  Joi,
+}: PluginOptionsSchemaArgs): JoiObject {
+  return Joi.object({
+    apiKey: Joi.string().required(),
+    password: Joi.string().required(),
+    storeUrl: Joi.string().required(),
```

Comment:
```suggestion
    storeUrl: Joi.string()
       .pattern(/^[a-z\-]+\.myshopify\.com$/)
       .message(`The storeUrl value should be your store's myshopify.com URL in the form "my-site.myshopify.com", without https or slashes`
       required(),
```
Commenting because I literally just got caught out my this: It would be good to have more specific formatting rules for the domain, because calling "URL" makes it seem like you should include the protocol etc, and doesn't make it clear that it should be the myshopify.com domain rather than the custom domain. (note, I haven't tested this suggestion). Changign the wording in the Cloud onboarding flow would be good too

Reply:
Good call @ascorbic . Thanks for the suggestion!